File: airbnb/Profile/src/main/java/com/example/demo/model/dto/UserDTO.java
package com.example.demo.model.dto;

import java.util.Set;

public class UserDTO {

    private String name;
    private String surname;
    private String email;

    private String username;

    private String password;
    private String telephone;
    private Set<String> roles;

    public UserDTO() {
    }

    public UserDTO(String name, String surname, String email, String username, String password, String telephone, Set<String> roles) {
        this.name = name;
        this.surname = surname;
        this.email = email;
        this.username = username;
        this.password = password;
        this.telephone = telephone;
        this.roles = roles;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getSurname() {
        return surname;
    }

    public void setSurname(String surname) {
        this.surname = surname;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getTelephone() {
        return telephone;
    }

    public void setTelephone(String telephone) {
        this.telephone = telephone;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public Set<String> getRoles() {
        return roles;
    }

    public void setRoles(Set<String> roles) {
        this.roles = roles;
    }
}

File: airbnb/Profile/src/main/java/com/example/profile/model/dtogrpc/UserDto.kt
package com.example.profile.model.dtogrpc

import com.example.profile.model.Role
import com.example.profile.model.User

data class UserDto @JvmOverloads constructor(
    val id: UserNameDto? = null,
    val name: String,
    val surname: String,
    val email: String,
    val password: String,
    val telephone: String,
    val roles: MutableSet<Role>
) {
    companion object {
        @JvmStatic
        fun convert(from: User): UserDto {
            return UserDto(
                from.id?.let{ UserNameDto.convert(it, from.username) },
                from.name,
                from.surname,
                from.email,
                from.password,
                from.telephone,
                from.roles
            )
        }
    }
}